source/module_hamilt_lcao/module_gint/grid_technique.h

```diff
@@ -20,98 +20,92 @@ class Grid_Technique : public Grid_MeshBall
+    int lgd;                   // local grid dimension.  lgd * lgd symmetry matrix.
+    std::vector<int> trace_lo; // trace local orbital.
 
     //---------------------------------------
     // nnrg: number of matrix elements on
     // each processor's real space grid.
     // use: GridT.in_this_processor
     //---------------------------------------
     int nnrg;
-    int* nlocdimg;
-    int* nlocstartg;
-
-    int* nad; // number of adjacent atoms for each atom.
-    int** find_R2;
-    int** find_R2_sorted_index;
-    int** find_R2st;
+    std::vector<int> nlocdimg;
+    std::vector<int> nlocstartg;
+    std::vector<int> nad; // number of adjacent atoms for each atom.
+    std::vector<std::vector<int>> find_R2;
+    std::vector<std::vector<int>> find_R2_sorted_index;
+    std::vector<std::vector<int>> find_R2st;
     bool allocate_find_R2;
     int binary_search_find_R2_offset(int val, int iat) const;
 
-    //UnitCell and LCAO_Obrbitals
+    // UnitCell and LCAO_Obrbitals
     const UnitCell* ucell;
     const LCAO_Orbitals* orb;
 
+    // UnitCell parameters
+    int nwmax;
+    int nr_max;
+    int ntype;
+
+    // LCAO Orbitals
+    double dr_uniform;
+		std::vector<double> rcuts;
+    std::vector<std::vector<double>> psi_u;
+    std::vector<std::vector<double>> dpsi_u;
+    std::vector<std::vector<double>> d2psi_u;
+
     // indexes for nnrg -> orbital index + R index
     std::vector<gridIntegral::gridIndex> nnrg_index;
 
-    // public functions
-  public:
-    Grid_Technique();
-    ~Grid_Technique();
-
-    void set_pbc_grid(const int& ncx_in,
-                      const int& ncy_in,
-                      const int& ncz_in,
-                      const int& bx_in,
-                      const int& by_in,
-                      const int& bz_in,
-                      const int& nbx_in,
-                      const int& nby_in,
-                      const int& nbz_in,
-                      const int& nbxx_in,
-                      const int& nbzp_start_in,
-                      const int& nbzp_in,
-                      const int& ny,
-                      const int& nplane,
-                      const int& startz_current,
-                      const UnitCell& ucell,
-                      const LCAO_Orbitals& orb,
-                      const int num_stream);
+    void set_pbc_grid(const int& ncx_in, const int& ncy_in, const int& ncz_in, const int& bx_in, const int& by_in,
+                      const int& bz_in, const int& nbx_in, const int& nby_in, const int& nbz_in, const int& nbxx_in,
+                      const int& nbzp_start_in, const int& nbzp_in, const int& ny, const int& nplane,
+                      const int& startz_current, const UnitCell& ucell, const double dr_uniform, 
+                      std::vector<double> rcuts,std::vector<std::vector<double>> psi_u,
+                      std::vector<std::vector<double>> dpsi_u,std::vector<std::vector<double>> d2psi_u
```

Comment:
maybe `rcuts`, `psi_u`, `dpsi_u` and `d2psi_u` should be passed by reference here.

Reply:
It has already been resolved.